tests/LexerTest.php

```diff
@@ -229,43 +238,6 @@ public function getStringWithEscapedDelimiter()
         ];
     }
 
-    /**
-     * @group legacy
-     * @dataProvider getStringWithEscapedDelimiterProducingDeprecation
-     */
-    public function testStringWithEscapedDelimiterProducingDeprecation(string $template, string $expected, string $expectedDeprecation)
```

Comment:
I think we should keep the tests to prove that double-escaped strings still work as expected.

Reply:
I did not remove those tests, I moved them to the `getStringWithEscapedDelimiter` above, see: https://github.com/twigphp/Twig/blob/ba4e8175cbd4c38928675ddcb0fd0f35361969d9/tests/LexerTest.php#L195-L206